### ISSUE TITLE
confirmation dialog requires newlines

### ISSUE THREAD

**Issue by ds26gte** [vim/vim#762]
When 'confirm' is on, and we're exiting vim with a bunch of unsaved files, the "Save changes" confirmation dialog requires a newline after the answer to every question but the first.

**Comment by brammool** [comment]
> When 'confirm' is on, and we're exiting vim with a bunch of unsaved
> files, the "Save changes" confirmation dialog requires a newline after
> the answer to every question but the first.

Can you give a step-by-step example?

## 

Experience is what you get when you don't get what you want.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\
\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by ds26gte** [comment]
- Start `vim -u NONE`
- Do `:set compatible confirm hidden`
- Edit three files, e.g.,
- `:e a.txt`
- `:e b.txt`
- `:e c.txt`
- Insert some text in all three of the files
- Seek to exit vim with `:qa`
- You will get a confirmation dialog:

```
Save changes to "a.txt"?
[Y]es, (N)o, Save (A)ll, (D)iscard All, (C)ancel:
```
- Type `y`. This will save `a.txt` and you will get a further confirmation dialog:

```
Save changes to “b.txt”?
[Y]es, (N)o, (C)ancel:
```
- Type `y`. **You will find that this is not enough! You will have to press the newline key too**, before you get the next confirmation dialog:

```
Save changes to “c.txt”?
[Y]es, (N)o, (C)ancel:
```
- Once again, you will have to type `y` followed by newline.

**Comment by brammool** [comment]
ds26gte wrote:

> - Start `vim -u NONE`
> - Do `:set compatible confirm hidden`
> - Edit three files, e.g.,
> - `:e a.txt`
> - `:e b.txt`
> - `:e c.txt`
> - Insert some text in all three of the files
> - Seek to exit vim with `:qa`
> - You will get a confirmation dialog:
> 
> ```
> Save changes to "a.txt"?
> [Y]es, (N)o, Save (A)ll, (D)iscard All, (C)ancel:
> ```
> - Type `y`. This will save `a.txt` and you will get a further confirmatio=
>   n dialog:
> 
> ```
> Save changes to =E2=80=9Cb.txt=E2=80=9D?
> [Y]es, (N)o, (C)ancel:
> ```
> - Type `y`. **You will find that this is not enough! You will have to pre=
>   ss the newline key too**, before you get the next confirmation dialog:
> 
> ```
> Save changes to =E2=80=9Cc.txt=E2=80=9D?
> [Y]es, (N)o, (C)ancel:
> ```
> - Once again, you will have to type `y` followed by newline.

Thanks, I updated the todo item.

## 

FATHER:       Make sure the Prince doesn't leave this room until I come and
              get him.
FIRST GUARD:  Not ... to leave the room ... even if you come and get him.
FATHER:       No.  Until I come and get him.
SECOND GUARD: Hic.
                 "Monty Python and the Holy Grail" PYTHON (MONTY) PICTURES LTD

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\
\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by lacygoill** [comment]
Fixed by [8.2.1147](https://github.com/vim/vim/releases/tag/v8.2.1147)

---

This todo item is no longer relevant:

> When repeating the 'confirm' dialog one needs to press Enter. (ds26gte, 2016
> Apr 17)  #762

https://github.com/vim/vim/blob/b80d2fb0e7b9adec4cff1bc4e2d30635cb70502e/runtime/doc/todo.txt#L1407-L1408
